src/sbt-test/settings/caching/test

```diff
@@ -0,0 +1,96 @@
+# Note that this scripted assumes that sbt-protoc uses "last modified" rather than a hash to
```

Comment:
I assume `setReadable` meant to modify the file timestamp? How about we call it `modifyFile` and would append a blank line or a line with `// modified!` - that would also decouple from the caching criteria.

Reply:
> I assume `setReadable` meant to modify the file timestamp?

Actually, it does not. It makes the file (in)accessible, without changing its "last modified" attribute. The goal is not to test how a change in input file triggers invalidation, but to instrument the input to detect whether it was actually consumed by `protoc` or not. That's why a hashing strategy would require revisiting that instriumentation strategy, as we wouldn't know if the access came from sbt-protoc or from protoc, thus this heading comment.

Reply:
I [already used this approach in the past](https://github.com/scalameta/sbt-scalafmt/pull/40/files#r310701679) as a blackbox way to tell whether a cached function consuming files was actually called or not.